instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -12,8 +12,10 @@ class CommentSection extends React.Component {
 
   addNewComment = (e) => {
     e.preventDefault();
+    const comments = this.state.comments.slice();
     if (e.target.commentInput.value) {
-      alert('NewComment');
+      comments.push({text: e.target.commentInput.value});
+      this.setState({comments: comments})
```

Comment:
Nice work here,  remember with es6 shorthand, if the key and value match, you only have to list it once.